File: C#/Programming Basics with C#/13. Nested Loops - Exercise/08. Train The Trainers/TrainTheTrainers.cs
using System;

class TrainTheTrainers
{
    static void Main()
    {
        int countJury = int.Parse(Console.ReadLine());
        double sumMultyEvalutions = 0;
        double averageEvalutionByPresentation = 0;
        double averageEvalutionByPresentations = 0;

        int counterPresentation = 0;

        for (string presentationName = Console.ReadLine(); presentationName != "Finish"; presentationName = Console.ReadLine())
        {
            double sumSingleEvalutions = 0;

            for (int i = 0; i < countJury; i++)
            {
                double input = double.Parse(Console.ReadLine());
                sumSingleEvalutions = sumSingleEvalutions + input;
            }

            averageEvalutionByPresentation = sumSingleEvalutions / countJury;
            sumMultyEvalutions = sumMultyEvalutions + averageEvalutionByPresentation;

            Console.WriteLine($"{presentationName} - {averageEvalutionByPresentation:F2}.");
            ++counterPresentation;
        }

        averageEvalutionByPresentations = sumMultyEvalutions / counterPresentation;
        Console.WriteLine($"Student's final assessment is {averageEvalutionByPresentations:F2}.");
    }
}

File: C++/Programming Basics with C++/13. Exercise - Nested loops/simple_task/TrainTheTrainers.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int main() {
    string input;
    getline(cin, input);
    int countJury = stoi(input);
    double sumMultiEvaluations = 0;
    double averageEvaluationByPresentation = 0;
    double averageEvaluationByPresentations = 0;

    int counterPresentation = 0;
    string presentationName;
    getline(cin, presentationName);
    cout.setf(ios::fixed);
    cout.precision(2);

    while (presentationName != "Finish") {
        double sumSingleEvaluations = 0;

        for (int i = 0; i < countJury; i++) {
            getline(cin, input);
            double num = stod(input);
            sumSingleEvaluations += num;
        }

        averageEvaluationByPresentation = sumSingleEvaluations / countJury;
        sumMultiEvaluations += averageEvaluationByPresentation;

        cout << presentationName << " - " << averageEvaluationByPresentation << "." << endl;
        ++counterPresentation;
        getline(cin, presentationName);
    }

    averageEvaluationByPresentations = sumMultiEvaluations / counterPresentation;
    cout << "Student's final assessment is " << averageEvaluationByPresentations << "." << endl;

    return 0;
}
